compiler/modules/front/resolveTypes.bal

```diff
@@ -258,7 +258,29 @@ function resolveTypeDesc(ModuleSymbols mod, s:ModuleLevelDefn modDefn, int depth
         }
     }
     if td is s:SingletonTypeDesc {
-        return t:singleton(mod.tc, td.value);
+        s:SimpleConstExpr valueExpr = td.valueExpr;
```

Comment:
A SimpleConstExpr is an Expr so you should just be able to handle this with resolveConstExpr.

Reply:
Fixed.